### ISSUE TITLE
Support PNG and GIF

### ISSUE THREAD

**Issue by idontgetoutmuch** [diagrams/diagrams-haddock#31]
I seem to have created an 11M SVG. It would be more economical to use a PNG. I'd also like to be able to embed animations (GIF).

**Comment by byorgey** [comment]
Yes, this makes sense.  I don't remember off the top of my head why we only support the `diagrams-svg` backend, or how much work it would be to generalize.

**Comment by byorgey** [comment]
After a quick perusal of the code I think it is mostly a symptom of a more general problem: it is currently quite difficult to write code which can choose different backends based on user input such as a flag.  I think this situation would probably improve with @cchalmers proposed changes to the way we organize backends and their types, although that may still be a ways off.

**Comment by idontgetoutmuch** [comment]
So a quick(ish) option would be to clone the repo and fix the backend to be rasterfic?

**Comment by byorgey** [comment]
Yes, I think that would probably work!
